### ISSUE TITLE
Set ceph.user.name before creating the ceph pool

### ISSUE THREAD

**Issue by dann1** [lxc/lxd#3993]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 16.04
 config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
api_status: stable
api_version: "1.0"
auth: trusted
public: false
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate:
  certificate_fingerprint: c4c22570f69bcbd3a748258a7863c9511c977655603c730b88f45656b2b433a2
  driver: lxc
  driver_version: 2.0.8
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.4.0-98-generic
  server: lxd
  server_pid: 6528
  server_version: "2.18"
  storage: ""
  storage_version: ""

# Issue description

Hello, I'm trying to bind lxd to a preexisting ceph cluster, I have succesfully configured the LXD node as a ceph client outside LXD scope. When I try to set ceph as storage backend for this LXD node via `lxd init` I'm not asked to specify a ceph.user.name, I need to specify a user that is not the default admin user. When I try `lxc storage set ceph.user.name=my_non_default_user`  it fails. 

# Steps to reproduce

 1. ```
root@anakin:/home/team/ceph# lxd init
Do you want to configure a new storage pool (yes/no) [default=yes]?
Name of the new storage pool [default=default]: ceph_cluster
Name of the storage backend to use (dir, btrfs, ceph, lvm) [default=btrfs]: ceph
Create a new CEPH pool (yes/no) [default=yes]? no
Name of the existing CEPH cluster [default=ceph]:
Name of the existing OSD storage pool [default=lxd]: one2
Would you like LXD to be available over the network (yes/no) [default=no]? no
Would you like stale cached images to be updated automatically (yes/no) [default=yes]? no
Would you like to create a new network bridge (yes/no) [default=yes]? no
error: Failed to run: ceph --name client.admin --cluster ceph osd pool create one2 32: 2017-11-03 11:31:56.087971 7f38dc070700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin: (2) No such file or directory
2017-11-03 11:31:56.090188 7f38dc070700 -1 monclient(hunting): authenticate NOTE: no keyring found; disabled cephx authentication
2017-11-03 11:31:56.090195 7f38dc070700  0 librados: client.admin authentication error (95) Operation not supported
Error connecting to cluster: Error
root@anakin:/home/team/ceph#
```

 2. `root@anakin:/home/team/ceph# lxc storage set ceph.user.name=oneadmin
error: wrong number of subcommand arguments`


# Information to attach
Also, that ceph cluster has been configurated with a defined placement group number. If I try to reutilize that ceph cluster will the default PG num lxd uses be in conflict with the configured PG num.

**Comment by dann1** [comment]
I have bound lxd to my previous ceph cluster, but there are some issues.
When I query ceph space status one2 pool has 2988G space
```
root@anakin:/home/team# ceph --id oneadmin df
GLOBAL:
    SIZE       AVAIL     RAW USED     %RAW USED
    13020G     9643G        3377G         25.94
POOLS:
    NAME     ID     USED      %USED     MAX AVAIL     OBJECTS
    rbd      0          0         0         2988G           0
    one      1      1130G     27.44         2988G      298179
    one2     2      2301M      0.08         2988G         602
```

But when I do  that via lxd one2 pool has only 27G
```
root@anakin:/home/team# lxc storage show ceph_lxd_cluster
config:
  ceph.cluster_name: ceph
  ceph.osd.pg_num: "512"
  ceph.osd.pool_name: one2
  ceph.user.name: oneadmin
  size: 27GB
  source: one2
  volatile.pool.pristine: "true"
description: ""
name: ceph_lxd_cluster
driver: ceph
used_by: []
```

Also when I try to instantiate a container it seems LXD fails to get the rootfs

```
root@anakin:/home/team# lxc launch ubuntu:
Creating the container
error: Failed container creation:           
 - https://cloud-images.ubuntu.com/releases: No root device could be found.
```

**Comment by dann1** [comment]
OK I think I did it, the rootfs issue was due to the rootfs disk inside container not pointing to the one2 ceph osd pool. I think that due to the creation of the ceph storage pool not done via `lxd init` the default container profile didn't get updated, so based on https://asciinema.org/a/135437 I added
 ```
devices:
  root:
    path: /
    pool: ceph_lxd_cluster
    type: disk
```
and things worked out fine.  I'm renaming the issue.
